src/libs/actions/Task.js

```diff
@@ -259,10 +265,26 @@ function completeTask(taskReportID, taskTitle) {
         },
     ];
 
-    // Update optimistic data for parent report action
+    // Update optimistic data and failure data for parent report action
     const optimisticParentReportData = ReportUtils.getOptimisticDataForParentReportAction(taskReportID, completedTaskReportAction.created, CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
     if (!_.isEmpty(optimisticParentReportData)) {
         optimisticData.push(optimisticParentReportData);
+        // Get assignee report from participantAccountIDs as in creating task
+        const assigneeChatReportID = lodashGet(ReportUtils.getChatByParticipants(taskReport.participantAccountIDs), 'reportID');
+        // We don't need to create optmistic report action for assignee report if assignee and shareDestination are the same
+        if (assigneeChatReportID && `${assigneeChatReportID}` !== `${taskReport.parentReportID}`) {
+            const parentReportAction = ReportActionsUtils.getParentReportActionForTask(taskReportID, assigneeChatReportID);
+            if (parentReportAction) {
+                optimisticData.push({
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${assigneeChatReportID}`,
+                    value: {
+                        // Copy optimistic report action data from shareDestination
+                        [parentReportAction.reportActionID]: _.values(optimisticParentReportData.value)[0],
```

Comment:
Accessing the first element of the array does not tell us much what are accessing. This can break easily if `optimisticParentReportData` changes.

Can we call `updateOptimisticParentReportAction` and use that report action to update both parents?

Reply:
Updated